docs/specification.md

```diff
@@ -59,10 +59,15 @@ data Config proposalMetadata = Config
   -- ^ Name of the DAO.
   , cDaoDescription :: Text
   -- ^ Description of the DAO.
-  , cProposalCheck :: Lambda (Natural, Proposal proposalMetadata) Bool
+  , cProposalCheck :: Lambda (Natural, Proposal proposalMetadata) (Maybe Natural)
   -- ^ A lambda used to verify whether a proposal can be submitted.
   -- It checks 2 things: the proposal itself and the amount of tokens frozen upon submission.
   -- It allows the DAO to reject a proposal by arbitrary logic and captures bond requirements
+  -- It return `None` if the proposal is accepted
+  -- It return `Just Natural` if the proposal is rejected. `Natural` refer to
+  -- the amount of token that will be slashed.
+  -- For example, if Alice freezes 100 tokens to vote and this lambda divides the value by 2,
+  -- only 50 tokens will be unfrozen and other 50 tokens will be burnt.
```

Comment:
I've updated this back to the way we originally thought about it, and simply add extra `Proposal` to `cRejectedProposalReturnValue` since we need access to `proposerFrozenToken` of this particular proposal to be able to apply this slash function.